lib/puppet/x/jenkins/provider/cli.rb

```diff
@@ -130,7 +130,9 @@ def self.cli(command, options = {}, cli_pre_cmd = [])
                          false
                        end
 
-    Puppet.debug("#{sname} stdin:\n#{input}")
+    # This Puppet.debug shows the JSON with changed credentials in plain text.
+    # ToDo: Remove this debug output OR mask the credentials
+    # Puppet.debug("#{sname} stdin:\n#{input}")
```

Comment:
If this contains credentials, you should note that it's written to a file just below and there's a `chmod 644` rescue block, which makes those credentials world readable.

It looks like that is designed for the case where the `jenkins` user is not yet available. I wonder if that workaround should be removed and instead hard fail.

It should also be noted that there's no way to ensure the file really is removed. If `execute_with_retry` raises an exception it's not removed. That means in some cases the credentials could remain world readable until `$TMPDIR` is cleaned.

Reply:
At least this would stay on the server and not be shipped off to something like The Foreman where many more can see the credentials.